### PR TITLE
allow adding and editing subscriptions with members that left a…

### LINKED ISSUE #544: Bug: can't edit sub with member that moved to other sub
If in sub 1 with members A and B, member A leaves (B stays and sub 1 stays active) and A joins sub 2, sub 1 can't be edited anymore in the Django admin. It will show the message, that member A joined another sub already.

### PR DESCRIPTION
**Pull request by pascalfree**
…nd joined other subscription

resolves #544

### CODE REVIEW

File: juntagrico/lifecycle/submembership.py [hunk 1]
```diff
@@ -45,20 +46,22 @@ def check_sub_membership_consistency(instance):
...
-    check_sub_membership_consistency_ms(member, subscription, instance.join_date)
+    check_sub_membership_consistency_ms(member, subscription, instance.join_date, instance.leave_date)
 
 
-def check_sub_membership_consistency_ms(member, subscription, asof):
-    from juntagrico.dao.subscriptionmembershipdao import SubscriptionMembershipDao
-    if subscription.state == 'waiting' and SubscriptionMembershipDao.get_other_waiting_for_member(member,
-                                                                                                  subscription).count() > 0:
+def check_sub_membership_consistency_ms(member, subscription, join_date, leave_date):
+    from juntagrico.entity.member import SubscriptionMembership
```
Comment (motu55) [inline juntagrico/lifecycle/submembership.py:53]: why not at the top of the file

**Review by motu55 — COMMENTED** [review]
do i see it correctly that an admin can selct any member and if this member is invalid to be added to an subscription an exception will be thrown?

File: juntagrico/lifecycle/submembership.py (hunk 1 above)
Comment (pascalfree) [reply juntagrico/lifecycle/submembership.py:53]: due to circular imports

**Comment by pascalfree** [comment]
Yes there will be an evaluation error that is shown in the admin.
As the evaluation depends on the join and leave dates (that might be changed in the same request) we can not select the available members before sending the complete form.

There is a related issue #394 that suggests that the member preselection should be removed completely.

**Review by motu55 — APPROVED** [review]
ok i see. the restrictions came from a time when we did not have the special member subscription mapping, and so a restriction is not possible any more.
but it would be nice if a member is added to a subscription the join date would be set at the earliest point possibe

**Pull request merged**